proposals/partial-events-and-constructors.md

```diff
@@ -0,0 +1,262 @@
+{
+    [Export("initWithFormat:packetCapacity:")]
+    public partial AVAudioCompressedBuffer(AVAudioFormat format, uint packetCapacity);
+
+    [Export("create:")]
+    public partial event EventHandler Created;
+}
+```
+
+And the source generator would generate the bindings (to Objective-C in this case):
+
+```cs
+partial class AVAudioCompressedBuffer : AVAudioBuffer
+{
+    [BindingImpl(BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
+    public partial AVAudioCompressedBuffer(AVAudioFormat format, uint packetCapacity) : base(NSObjectFlag.Empty)
+    {
+        // Call Objective-C runtime:
+        InitializeHandle(
+            global::ObjCRuntime.NativeHandle_objc_msgSendSuper_NativeHandle_UInt32(
+                this.SuperHandle,
+                Selector.GetHandle("initWithFormat:packetCapacity:"),
+                format.GetNonNullHandle(nameof(format)),
+                packetCapacity),
+            "initWithFormat:packetCapacity:");
+    }
+
+    public partial event EventHandler Created
+    {
+        add { /* ... */ }
+        remove { /* ... */ }
+    }
+}
+```
+
+## Detailed design
+
+### General
+
+Event declaration syntax ([§15.8.1][event-syntax]) is extended to allow the `partial` modifier:
+
+```diff
+ event_declaration
+-    : attributes? event_modifier* 'event' type variable_declarators ';'
++    : attributes? event_modifier* 'partial'? 'event' type variable_declarators ';'
+-    | attributes? event_modifier* 'event' type member_name
++    | attributes? event_modifier* 'partial'? 'event' type member_name
+         '{' event_accessor_declarations '}'
+     ;
+```
+
+Instance constructor declaration syntax ([§15.11.1][ctor-syntax]) is extended to allow the `partial` modifier:
+
+```diff
+ constructor_declaration
+-    : attributes? constructor_modifier* constructor_declarator constructor_body
++    : attributes? constructor_modifier* 'partial'? constructor_declarator constructor_body
+     ;
+```
+
+Note that there is [a proposal][partial-ordering] to allow the `partial` modifier anywhere among modifiers,
+rather than only as the last one (also for method, property, and type declarations).
+
+An event declaration with the `partial` modifier is said to be a *partial event declaration*
+and it is associated with one or more *partial events* with the specified names
+(note that one event declaration without accessors can define multiple events).
+
+A constructor declaration with the `partial` modifier is said to be a *partial constructor declaration*
+and it is associated with a *partial constructor* with the specified signature.
+
+A partial event declaration is said to be an *implementing declaration*
+when it specifies the `event_accessor_declarations` or it has the `extern` modifier.
+Otherwise, it is a *defining declaration*.
+
+A partial constructor declaration is said to be a *defining declaration*
+when it has the semicolon body and it lacks the `extern` modifier.
```

Comment:
>the [](http://example.com/codeflow?start=12&length=3)

"a"?

Reply:
I assume you are referring to "the semicolon body", not "the `extern` modifier" here?